.github/workflows/publish-package.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish to registry
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    if: "contains(github.event.head_commit.message, '[release]')"
```

Comment:
I think it's a good idea to lock this pipeline to just run on commits made by the concourse worker, will update.

Reply:
it's tricker to do this than appears as it will always be the case that someone else will be merging the PR, not the concourse user. This means that the head_commit author will be the merging user and not the concourse user.